### PR TITLE
ansible-inventory: rework generation of groups

### PR DESCRIPTION
**Pull request by pvoborni**
The Ansible inventory output was not correctly adding hosts into groups. E.g.
group was not added when it was not defined in layout. Nested group was not
correctly found and host added.

This part is completely reworked and changes the behavior to:
- add only a host name reference (no vars) into custom groups
- add missing groups as children of 'all' group
- works with both "groups" and "group" defined in metadata
- add all hosts to 'all' with also variables. This ensures that the full host
  object is defined only once.

Doc: https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html

Signed-off-by: Petr Vobornik <pvoborni@redhat.com>

### CODE REVIEW

File: tests/unit/test_ansible_inventory.py
Comment (netoarmando) [inline tests/unit/test_ansible_inventory.py:69]: You can move this to a fixture called `inventory`; (You can do the same to `metadata` and `db`) Pass them to `test_common_layout` and move the `ensure_*` functions to become methods.

Fixtures are the best practice to setup and teardown scenarios.

Comment (pvoborni) [reply tests/unit/test_ansible_inventory.py]: Amended and several tests added (through parametrize).

File: src/mrack/outputs/ansible_inventory.py
Comment (Tiboris) [inline src/mrack/outputs/ansible_inventory.py:77]: Just a note that it does not return `False`/`True` Could you please update the doctext?

**Review by Tiboris — CHANGES_REQUESTED** [review]
Thanks for the update @pvoborni changes looks good to me, but please check the inline comment.
Let us wait for @netoarmando to give final review.

**Review by netoarmando — APPROVED** [review]
LGTM! I believe the solution for parametrizing the fixtures is good enough!

**Review by Tiboris — APPROVED** [review]

**Pull request merged**
